### PR TITLE
fix: remove do you work in

### PR DESCRIPTION
**Pull request by ColinBuyck**
This PR addresses #678 

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

This PR removes every reference to workInRegion and subsequently, work address in sites/public, sites/partners, and most of shared-helpers.
Exceptions:

- The backend-swagger.ts file and api folder were not altered.
- testHelpers.ts and applicationData.ts for cypress mock data employ the backend application type which requires the applicantWorkAddress field. 

This approach was discussed with @ludtkemorgan as a way to avoid unnecessary tech debt by only hiding the field from the UI but keeping the backend functionality unchanged. It will likely cause some merge conflicts in the public and partner's folders but given the field doesn't really match the unique approach to jurisdictions that Doorway has shifted to, I believe it is necessary.

## How Can This Be Tested/Reviewed?

This can be tested by trying to break the common app flow, verifying that these questions don't appear on the partners side (paper app add, edit, or view), and general testing around applications.

## Author Checklist:

- [x] Added QA notes to the issue with applicable URLs
- [x] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [ ] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

File: sites/partners/page_content/locale_overrides/general.json [hunk 1]
```diff
@@ -63,6 +60,7 @@
   "applications.addConfirmModalAddApplicationPostLotteryWeighted": "Adding a new application will require the lottery to be re-run.",
   "applications.addConfirmModalContent": "This listing has closed, are you sure you want to add an application?",
   "applications.addConfirmModalHeader": "Confirmation needed",
+  "application.details.vouchers": "Housing Voucher or Subsidy",
```
Comment (mcgarrye) [inline sites/partners/page_content/locale_overrides/general.json:63]: Any reason you are moving this?

File: sites/partners/src/components/applications/PaperApplicationForm/sections/FormPrimaryApplicant.tsx
Comment (mcgarrye) [inline sites/partners/src/components/applications/PaperApplicationForm/sections/FormPrimaryApplicant.tsx:1]: There are some imports that are now unused in this file that can be cleaned up, ie FieldValue and YesNoEnum

File: sites/partners/src/components/applications/PaperApplicationDetails/sections/DetailsPrimaryApplicant.tsx
Comment (mcgarrye) [inline sites/partners/src/components/applications/PaperApplicationDetails/sections/DetailsPrimaryApplicant.tsx:1]: Same here with import of YesNoEnum

File: sites/partners/src/components/applications/PaperApplicationDetails/DetailsAddressColumns.tsx
Comment (mcgarrye) [inline sites/partners/src/components/applications/PaperApplicationDetails/DetailsAddressColumns.tsx:1]: YesNoEnum as well

**Review by mcgarrye — APPROVED** [review]
Woohooooo

**Comment by ColinBuyck** [comment]
Tidied up, thanks for the review @mcgarrye 🌵

**Review by ludtkemorgan — APPROVED** [review]

**Pull request merged**